numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1120,6 +1120,30 @@ arr_unravel_index(PyObject *self, PyObject *args, PyObject *kwds)
         goto fail;
     }
 
+    if (!PyArray_CanCastArrayTo(indices, dtype, NPY_SAFE_CASTING)) {
```

Comment:
Indeed, I see that it wouldn't work all that well with certain other types, as the Travis failures noticed.